builder/azure/arm/config.go

```diff
@@ -195,6 +196,11 @@ type Config struct {
 	// If set to true, Virtual Machines deployed from the latest version of the
 	// Image Definition won't use this Image Version.
 	SharedGalleryImageVersionExcludeFromLatest bool `mapstructure:"shared_gallery_image_version_exclude_from_latest" required:"false"`
+	// Specify the storage account type for gallery image version.
+	// Valid values are Standard_LRS and Standard_ZRS.
+	// The default is Standard_LRS.
+	SharedGalleryImageVersionStorageAccountType string `mapstructure:"shared_gallery_image_version_storage_account_type" required:"false"`
+	sharedGalleryImageVersionStorageAccountType newCompute.StorageAccountType
```

Comment:
What's the reasoning between having an exported and an unexported field here? 

It looks like you're trying to preserve the raw user input but I'm not sure that's really necessary; we can do a validation check against the SharedGalleryImageVersionStorageAccountType and then just continue to use that, casting it to a newCompute.StorageAccountType later if we need to.

Reply:
Ah yea I was more or less following the method I saw for the `ManagedImageStorageAccountType` using type `compute.StorageAccountTypes`, but yea I agree I think casting may just be the better approach

Reply:
yeah I think we should just cast and then I think we can also avoid the switch I called out below